src/main/scala/encry/network/peer/PeerManager.scala

```diff
@@ -66,14 +66,12 @@ class PeerManager extends Actor with Logging {
         else None
       }
 
-      if (connectedPeers.size + connectingPeers.size < settings.network.maxConnections) {
+      if (connectedPeers.size + connectingPeers.size < settings.network.maxConnections)
         randomPeer.foreach { address =>
           if (!connectedPeers.exists(_._1 == address) &&
-            !connectingPeers.exists(_.getHostName == address.getHostName)) {
+            !connectingPeers.exists(_.getHostName == address.getHostName) && checkPossibilityToAddPeer(address))
```

Comment:
We can check these conditions while by adding `.filter` in `val peers: Seq[InetSocketAddress] = PeerDatabase.knownPeers().keys.toSeq` line